blog/views.py

```diff
@@ -0,0 +1,102 @@
+from django.views.generic import ListView, DetailView, \
+   CreateView, UpdateView, DeleteView
+from .models import Post, Category
+from .forms import PostForm, EditForm
+from django.urls import reverse_lazy
+from django.views import generic
+from django.contrib.auth.forms import UserCreationForm
+
+
+class HomeView(ListView):
+    model = Post
+    template_name = 'home.html'
+    ordering = ['-post_date']
+
+# TODO: move outside
+    def get_context_data(self, *args, **kwargs):
+        cat_menu = Category.objects.all()
+        context = super(HomeView, self).get_context_data()
+        context["cat_menu"] = cat_menu
+        return context
+
+
+class CategoryView(DetailView):
+    model = Post
+    template_name = 'categories.html'
+
+    def get_context_data(self, *args, **kwargs):
+        category_posts = Post.objects.filter(category=self.kwargs.get('pk'))
+        context = super(CategoryView, self).get_context_data()
+        context["category_posts"] = category_posts
+        return context
+
+
+class UserRegisterView(generic.CreateView):
+    form_class = UserCreationForm
+    template_name = 'registration/register.html'
+    success_url = reverse_lazy('login')
+
+    def get_context_data(self, *args, **kwargs):
+        cat_menu = Category.objects.all()
+        context = super(UserRegisterView, self).get_context_data()
+        context["cat_menu"] = cat_menu
+        return context
+
+
+class PostDetailView(DetailView):
+    model = Post
+    template_name = 'post_details.html'
+
+    def get_context_data(self, *args, **kwargs):
+        cat_menu = Category.objects.all()
+        context = super(PostDetailView, self).get_context_data()
+        context["cat_menu"] = cat_menu
```

Comment:
What is cat_menu? 
Please use more obvious names for variables, like 'category_list'. 
Also you can rewrite lines 51 and 53 to one line and make code more clean

Reply:
done